src/plugins/dashboard/public/dashboard_top_nav/internal_dashboard_top_nav.tsx

```diff
@@ -313,10 +323,57 @@ export function InternalDashboardTopNav({
       });
     }
     if (showWriteControls && managed) {
-      allBadges.push(getManagedContentBadge(dashboardManagedBadge.getTooltip()));
+      allBadges.push({
+        renderCustomBadge: ({ badgeText }) => {
+          const badgeProps = {
+            ...getManagedContentBadge(badgeText, false),
+            onClickAriaLabel: dashboardManagedBadge.getAriaLabel(),
+            iconOnClick: () => setIsPopoverOpen(!isPopoverOpen),
+            iconOnClickAriaLabel: dashboardManagedBadge.getAriaLabel(),
+            onClick: () => setIsPopoverOpen(!isPopoverOpen),
+          } as EuiBadgeProps;
```

Comment:
To simplify this a tiny bit more, I think we can do the following:
1. Move the definition of `badgeProps` to be **before** you push to `allBadges`
2. Once you have that, everywhere you use `dashboardManagedBadge.getManagedText()`, you can replace it with `badgeProps.badgeText` - since `getManagedContentBadge` **already provides** an `i18n` "managed" string, I think it's best if we reuse that just incase we decide to change the label at some point
3.